curriculum/challenges/english/25-front-end-development/lab-magazine-layout/6735a7370e0ae93a4577c689.md

```diff
@@ -0,0 +1,228 @@
+---
+id: 6735a7370e0ae93a4577c689
+title: Design a Magazine Layout Using CSS Grid
+challengeType: 14
+dashedName: design-a-magazine-layout-using-css-grid
+# demoType can either be 'onClick' or 'onLoad'. If the project or lab doesn't have a preview, delete the property
+demoType: onClick
+---
+
+# --description--
+In this lab, you will design a magazine layout using CSS Grid, including concepts like grid rows and grid columns.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have a `main` element with the class `magazine-cover`.
+
+2. Your `.magazine-cover` should contain a `header` element for the title and `section` elements for each article and the cover image.
+
+3. You should set the display property of the `.magazine-cover` to `grid.`
+
+4. The `.magazine-cover` should define a grid layout with the following structure:
+
+   - The title spanning the top row.
+   - Two feature articles and a cover image in the second row.
+   - Two small articles and the cover image in the third row.
+
+5. The `.magazine-cover` should have a `grid-template-columns` property that divides the space into three equal parts.
+
+6. The `.magazine-cover` should have a `grid-template-rows` property. It should be set to `auto` for the first row, and divide equally the space occupied by the remaining two rows.
+
+7. You should add a gap between grid items and set the width of the `.magazine-cover` to occupy most of the viewport width.
+
+**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
+
+# --hints--
+
+You should have a `main` element.
+
+```js
+assert.exists(document.querySelector('main'));
+```
+
+You should have a `.magazine-cover` class.
```

Comment:
This is not checking that the main element has the requested class

Reply:
that is being checked in the next step